src/Lean/Server/Rpc/RequestHandling.lean

```diff
@@ -35,12 +36,19 @@ private def handleRpcCall (p : Lsp.RpcCallParams) : RequestM (RequestTask Json)
     fun snap => do
       if let some proc := (← builtinRpcProcedures.get).find? p.method then
         proc.wrapper p.sessionId p.params
-      else if let some proc := userRpcProcedures.getState snap.env |>.find? p.method then
-        proc.wrapper p.sessionId p.params
+      else if let some procName := userRpcProcedures.find? snap.env p.method then
+        let options := snap.cmdState.scopes.head!.opts
+        let proc : Except _ _ := Lean.Environment.evalConstCheck RpcProcedure snap.env options ``RpcProcedure procName
```

Comment:
Could it be a problem that we `evalConstCheck` on every RPC call? Supposing the imported .olean has an associated shared library, the interpreter would jump to fast native code here, right?